build.gradle

```diff
@@ -185,8 +195,23 @@ tasks.named('compileKotlin') {
     compileKotlin.classpath += files(compileGroovy.destinationDirectory)
 }
 
-test {
+tasks.named('test') {
   testLogging {
     events = ["standard_out", "standard_error", "started", "passed", "failed", "skipped"]
   }
+
+  // Hack for gradle runner test classloader.
+  //
+  // GradleRunner.withPluginClasspath method loads a plugin under test to the classloader_1.
+  // When the test project is executed, it will load plugins to a project classloader_2.
+  // The classloader_1 doesn't known about classloader_2. When the plugin under test will use classes which
+  // was loaded to project with the classloader_2, ClassNotFoundException will be thrown.
+  //
+  // To fix this problem, don't use the withPluginClasspath method.
+  // Provide a plugin under test with the maven local repository.
+  systemProperty("protobufPluginVersion", version)
+  dependsOn(
+    "publishProtobufPluginPluginMarkerMavenPublicationToMavenLocal",
```

Comment:
I'm uncomfortable using maven local, as it changes shared state and is not impacted by `gradle clean`. Could we instead define a repository that outputs into the build/ directory instead, by passing a local file as the repository URL? https://github.com/grpc/grpc-java/blob/6904f88a7d39337fd51659bfb049c9893b7070d4/build.gradle#L385 We can create a repository just for this purpose, which seems shouldn't be too hard.